### PR TITLE
update miner_tests

### PR DESCRIPTION
**Pull request by sidhujag**
Remove magic nonce's and reuse TestChain100Setup network.

Now no need to manually increment chaintip height as I call
ProcessBlock() after CreateBlock().

The FIXME's were fixed so CheckFinalTX(tx1) and CheckFinalTx(tx2) work.

### CODE REVIEW

File: src/test/miner_tests.cpp [hunk 1]
```diff
@@ -18,93 +18,33 @@
...
-// NOTE: These tests rely on CreateNewBlock doing its own self-validation!
+BOOST_FIXTURE_TEST_SUITE(miner_tests, TestChain100Setup)
 BOOST_AUTO_TEST_CASE(CreateNewBlock_validity)
 {
-    CScript scriptPubKey = CScript() << ParseHex("04678afdb0fe5548271967f1a67130b7105cd6a828e03909a67962e0ea1f61deb649f6bc3f4cef38c4f35504e51ec112de5c384df7ba0b8d578a4c702b6bf11d5f") << OP_CHECKSIG;
+    CScript scriptPubKey = CScript() <<  ToByteVector(coinbaseKey.GetPubKey()) << OP_CHECKSIG;
     CBlockTemplate *pblocktemplate;
     CMutableTransaction tx,tx2;
     CScript script;
+	std::vector<unsigned char> vchSig;
```
Comment (sipa) [inline src/test/miner_tests.cpp]: Indentation?

Comment (sidhujag) [inline src/test/miner_tests.cpp]: weird, I use Visual Studio's as my editor and its showing just fine. I think /t and spaces mixed together may be causing problems? want me to update?

Comment (sipa) [inline src/test/miner_tests.cpp]: Yes, indentation is 4 spaces, nothing else. Thanks!

**Comment by laanwj** [comment]
Concept ACK, but please squash and use an informative commit message.

File: src/test/test_bitcoin.cpp [hunk 2]
```diff
@@ -116,10 +116,13 @@ TestChain100Setup::CreateAndProcessBlock(const std::vector<CMutableTransaction>&
     CBlockTemplate *pblocktemplate = CreateNewBlock(scriptPubKey);
     CBlock& block = pblocktemplate->block;
 
-    // Replace mempool-selected txns with just coinbase plus passed-in txns:
-    block.vtx.resize(1);
-    BOOST_FOREACH(const CMutableTransaction& tx, txns)
-        block.vtx.push_back(tx);
+    // Replace mempool-selected txns with just coinbase plus passed-in txns if txns is non-empty:
```
Comment (Diapolo) [inline src/test/test_bitcoin.cpp]: Nit: Indentation

**Comment by laanwj** [comment]
Needs rebase + nits addressed

**Comment by laanwj** [comment]
Closing due to inactivity; let me know if you start work on this again

**Pull request closed without merging**